docs/content/docs/usage/intl.md

```diff
@@ -40,183 +40,101 @@ Now create differents `.yml` files in a `translations` folder.
+
+```console
+npm i -S flat
+```
 
 We will assume all your urls will contain the locale key as the first part
+So let's initialize `react-intl` by creating an `src/utils/intl.js` file.
 
 ```
 http://domain.tld/en
 http://domain.tld/fr
 ```
 
-You can read the locale like this in your appContainer using ``location`` from
-react-router context
-
-```js
-const locale = location.pathname.replace(/^\//, "").split("/")[0]
-```
-
-For you homepage, you can choose to rely on a default locale in your code.
-
-This should be enough in most cases.
-
-When you have the locale, you can rely on it in ``AppContainer`` to initialize
-[``IntlProvider``](https://github.com/yahoo/react-intl/wiki/Components#intlprovider)
-
-// @todo document more
-
-That said, you might want something more complex, you can use the following
-section.
-
----
-
-## Using Redux
-
-### Create redux action
-
-We will define our `redux` actions constants.
-Create an `actions.js` file under a `constants` folder.
-
-```js
-export const SET_LOCALE = "SET_LOCALE"
-```
-
-### Create intl actions
-
-We need to implement the `setLocale` `redux` action.
-
-We will need the `flat` lib to transform our translations files into flatten objects
-
-```console
-npm i -S flat
-```
-
-Create a `intl.js` file under an `actions` folder.
-
 ```js
 import {addLocaleData} from "react-intl"
-import flatten from "flat"
-
-// Import locales you need to handle
 import __intlEN from "react-intl/locale-data/en"
 import __intlFR from "react-intl/locale-data/fr"
+import flatten from "flat"
 
-// Import your translations files
+// Get our translations files
 import localeEN from "translations/en.yml"
 import localeFR from "translations/fr.yml"
 
-import {SET_LOCALE} from "constants/actions"
-
+// Load "en" and "fr" utils
 addLocaleData(__intlEN)
 addLocaleData(__intlFR)
 
-function compileLocale(locale, {messages, formats = {}}) {
-  return {locale, messages: flatten(messages), formats}
+const locales = ["fr", "en"]
+const defaultLocale = "en"
+const messages = {
+  en: flatten(localeEN),
+  fr: flatten(localeFR),
 }
 
-const LOCALES = {
-  "en": compileLocale("en", localeEN),
-  "fr": compileLocale("fr", localeFR),
-}
-
-export function setLocale(locale) {
-  window.localStorage.setItem("locale", locale)
-  return {type: SET_LOCALE, ...LOCALES[locale]}
-}
-```
-
-### Create intl reducer
-
-Create a `intl.js` file under a `reducers` folder
-
-```js
-import {SET_LOCALE} from "constants/actions"
-
-const initialState = {
-  locale: null,
-  messages: null,
-  formats: null,
-}
-
-export default function intlReducer(state = initialState, action) {
-  if (action.type === SET_LOCALE) {
-    return {
-      locale: action.locale,
-      messages: action.messages,
-      formats: action.formats,
-    }
+// We will inject this into IntlProvider
+export function getIntl(location) {
+  const firstURIlevel = location.pathname.replace(/^\//, "").split("/")[0]
```

Comment:
Can you add a comment on this line to explain what it does? Just so people quickly understand this tricky line ^^